lib/puppet/provider/package/snap.rb

```diff
@@ -7,109 +7,100 @@
   desc "Package management via Snap.
 
     This provider supports the `install_options` attribute, which allows snap's flags to be
-    passed to Snap. Namely `classic`, `dangerous`, `devmode`, `jailmode`, `channel`."
+    passed to Snap. Namely `classic`, `dangerous`, `devmode`, `jailmode`.
+
+    The 'channel' install option is deprecated and will be removed in a future release.
+  "
 
   commands snap_cmd: '/usr/bin/snap'
-  has_feature :installable, :install_options, :uninstallable, :purgeable
+  has_feature :installable, :versionable, :install_options, :uninstallable, :purgeable
   confine feature: %i[net_http_unix_lib snapd_socket]
 
   def self.instances
-    instances = []
-    snaps = installed_snaps
-
-    snaps.each do |snap|
-      instances << new(name: snap['name'], ensure: snap['version'], provider: 'snap')
+    installed_snaps.map do |snap|
+      new(name: snap['name'], ensure: snap['channel'], provider: 'snap')
     end
-
-    instances
   end
 
   def query
-    instances = self.class.instances
-    instances.each do |instance|
-      return instance if instance.name == @resource[:name]
-    end
-
-    nil
+    self.class.instances.find { |it| it.name == @resource[:name] }
   end
 
   def install
-    self.class.modify_snap('install', @resource[:name], @resource[:install_options])
+    modify_snap('install', @resource[:install_options])
   end
 
   def update
-    self.class.modify_snap('refresh', @resource[:name], @resource[:install_options])
+    modify_snap('refresh', @resource[:install_options])
   end
 
   def latest
-    params = URI.encode_www_form(name: @resource[:name])
-    res = PuppetX::Snap::API.get("/v2/find?#{params}")
-
-    raise Puppet::Error, "Couldn't find latest version" if res['status-code'] != 200
-
-    # Search latest version for the specified channel. If channel is unspecified, fallback to latest/stable
-    channel = if @resource[:install_options].nil?
-                'latest/stable'
-              else
-                self.class.parse_channel(@resource[:install_options])
-              end
-
-    selected_channel = res['result'].first&.dig('channels', channel)
-    raise Puppet::Error, "No version in channel #{channel}" unless selected_channel
-
-    # Return version
-    selected_channel['version']
+    query['channel']
```

Comment:
Revert this, we need to query the Store for the latest version, not what we have locally.